data/moves.ts

```diff
@@ -13775,10 +13775,23 @@ export const Moves: {[moveid: string]: MoveData} = {
 				move.infiltrates = true;
 			}
 		},
-		onHit(target, source) {
+		onTryMove(source, target, move) {
+			if (source.isAlly(target) && source.volatiles['healblock']) {
+				this.attrLastMove('[still]');
+				this.add('cant', source, 'move: Heal Block', move);
+				return false;
+			}
+		},
+		onHit(target, source, move) {
 			if (source.isAlly(target)) {
 				if (!this.heal(Math.floor(target.baseMaxhp * 0.5))) {
-					this.add('-immune', target);
+					if (target.volatiles['healblock'] && target.hp !== target.maxhp) {
+						this.attrLastMove('[still]');
+						// Wrong error message, correct one not supported yet
+						this.add('cant', source, 'move: Heal Block', move);
+					} else {
+						this.add('-immune', target);
+					}
```

Comment:
Since we don't have general support for the Heal Block failure messages for things like, using Heal Pulse into a target affected by Heal Block either, I agree generalizing this is the way to go. I'll have to check all kinds of healing to make sure that Heal Block actually prevents it loudly every time, though.

I have no idea how to actually do that, though. onTryHeal has 2 sets of args atm and both expect boolean return values. I don't really want to refactor all of healing to fix this interaction myself.